### PR TITLE
creates and solves tests for ticket #11

### PR DESCRIPTION
**Pull request by clipeus-virtutis**
creates tests and solves for ticket #11

-  status(201), responds with the new comment
- status(400), Bad Request message when a username doesn't exist
- status(400), Bad Request message when required properties aren't provided
- status(400), Bad Request message when properties are the wrong data type
- status(404), responds with a 'Not Found' message when article_id doesn't exist

### CODE REVIEW

File: models/article-models.js [hunk 1]
```diff
@@ -39,6 +39,24 @@ exports.fetchComments = (articleId) => {
...
+  const votes = 0;
+  const intArticleId = parseInt(articleId);
+  let createdAt = Date.now();
+  const inputTime = convertTimestampToDate({ created_at: createdAt });
+  return db
+    .query(
+      `
+    INSERT INTO comments (body, votes, author, article_id, created_at)
+    VALUES ($1, $2, $3, $4, $5) RETURNING *;`,
+      [body, votes, username, intArticleId, inputTime.created_at]
```
Comment (dav3rid) [inline models/article-models.js]: Take a look at the seed function - you may find the votes and created_at are already handled for new insertions!

File: controllers/article-controllers.js [hunk 2]
```diff
@@ -46,3 +47,15 @@ exports.getComments = (req, res, next) => {
     })
     .catch(next);
 };
+
+exports.postComment = (req, res, next) => {
+  const { article_id } = req.params;
+  // console.log(article_id, "arty ID");
+  // console.log(req.body, "BODY");
+
+  Promise.all([checkArticleExists(article_id), addComment(article_id, req.body)])
```
Comment (dav3rid) [inline controllers/article-controllers.js]: Shouldn't need to check if the article exists here - PSQL will shout at us if the article id is not in the articles table due to the FOREIGN KEY constraint

File: models/article-models.js (hunk 1 above)
Comment (clipeus-virtutis) [reply models/article-models.js]: thanks Dave, saw the manage tables.js file and can now see that the comments table has a couple of default values for votes and created_at!

File: controllers/article-controllers.js (hunk 2 above)
Comment (clipeus-virtutis) [reply controllers/article-controllers.js]: updated test to 400 bad request instead of 404 not found and amended model to not check for article existing

**Pull request merged**